pkg/index/validate.go

```diff
@@ -90,8 +90,13 @@ func (p Platform) Validate() error {
 	if p.Bin == "" {
 		return errors.New("bin has to be set")
 	}
-	if len(p.Files) == 0 {
-		return errors.New("can't have a plugin without specifying file operations")
+	for _, fo := range p.Files {
```

Comment:
Let's extract this to a new method that we can unit-test, just like ValidatePlatform above.